src/docs/content/tutorials/scripting-intro/code/BasicSimulationJava.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2011-2023 GatlingCorp (https://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//#setup-the-file
+
+    package computerdatabase;
+
+    import static io.gatling.javaapi.core.CoreDsl.*;
+    import static io.gatling.javaapi.http.HttpDsl.*;
+
+    import io.gatling.javaapi.core.*;
+    import io.gatling.javaapi.http.*;
+
+//#setup-the-file
+
+//#extend-the-simulation-class
+
+public class ComputerDatabaseSimulation extends Simulation {
+
+
+}
+
+//#extend-the-simulation-class
+
+//#define-the-protocol-class
+
+public class ComputerDatabaseSimulation extends Simulation {
```

Comment:
In Java, you can't have multiple classes at the root of the same file. Your Java code doesn't compile.
You would have to wrap everything into one root class, eg:

```java
class Foo {
public class ComputerDatabaseSimulation extends Simulation {

}

public class ComputerDatabaseSimulation extends Simulation {

}

}

```

or, you can split into multiple files